src/overlays/actors/ovl_En_An/z_en_an.c

```diff
@@ -13,160 +20,3444 @@
+
+        this->stateFlags &= ~ENAN_STATE_ENGAGED;
+        this->stateFlags |= ENAN_STATE_LOST_ATTENTION;
+        this->loseAttentionTimer = 20;
+        this->msgScriptResumePos = 0;
+        this->actionFunc = EnAn_FollowSchedule;
+
+        return;
+    }
+
+    if ((this->scheduleResult == ANJU_SCH_WAITING_FOR_KAFEI) || (this->scheduleResult == ANJU_SCH_LAUNDRY_POOL_SIT) ||
+        (this->scheduleResult == ANJU_SCH_RANCH) || (this->scheduleResult == ANJU_SCH_COOKING) ||
+        (this->scheduleResult == ANJU_SCH_WITH_KAFEI)) {
+        return;
+    }
+
+    if ((this->lookAtActor != NULL) && (this->lookAtActor->update != NULL)) {
+        // Rotate towards actor
+        s16 yaw;
+        Vec3f lookAtPos;
+        Vec3f pos;
+
+        Math_Vec3f_Copy(&lookAtPos, &this->lookAtActor->world.pos);
+        Math_Vec3f_Copy(&pos, &this->actor.world.pos);
+        yaw = Math_Vec3f_Yaw(&pos, &lookAtPos);
+
+        Math_ApproachS(&this->actor.shape.rot.y, yaw, 4, 0x2AA8);
+    }
+}
+
+void EnAn_HandleCouplesMaskCutscene(EnAn* this, PlayState* play) {
+    s32 animIds[] = {
+        /* 0 */ 0, // no cue
+        /* 1 */ ENAN_ANIM_SITTING_IN_DISBELIEF,
+        /* 2 */ ENAN_ANIM_LOOK_UP,
+        /* 3 */ ENAN_ANIM_MASK_STAND_LOOP,
+        /* 4 */ ENAN_ANIM_MASK_KNEEL,
+        /* 5 */ ENAN_ANIM_HUG,
+        /* 6 */ ENAN_ANIM_HUG_RELEASE,
+        /* 7 */ ENAN_ANIM_COMBINE_MASKS_1,
+        /* 8 */ ENAN_ANIM_COMBINE_MASKS_2,
+        /* 9 */ ENAN_ANIM_HOLD_HANDS,
+    };
+    s32 pad;
+
+    //! FAKE
+    if (0) {}
+
+    if (Cutscene_IsCueInChannel(play, CS_CMD_ACTOR_CUE_557)) {
+        s32 cueChannel = Cutscene_GetCueChannel(play, CS_CMD_ACTOR_CUE_557);
+        u16 cueId = play->csCtx.actorCues[cueChannel]->id;
+
+        if (this->cueId != (cueId & 0xFF)) {
+            this->cueId = cueId;
+
+            if (this->cueId == 3) {
+                SET_WEEKEVENTREG(WEEKEVENTREG_COUPLES_MASK_CUTSCENE_STARTED);
+                this->drawMoonMask = true;
+            }
+            if (this->cueId == 9) {
+                this->drawMoonMask = false;
+            }
+
+            EnAn_ChangeAnim(this, play, animIds[cueId]);
+        }
+
+        if ((this->animIndex == ENAN_ANIM_MASK_KNEEL) || (this->animIndex == ENAN_ANIM_HUG) ||
+            (this->animIndex == ENAN_ANIM_HUG_RELEASE) || (this->animIndex == ENAN_ANIM_LOOK_UP)) {
+            if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+                EnAn_ChangeAnim(this, play, this->animIndex + 1);
+            }
+        }
+
+        Cutscene_ActorTranslateAndYaw(&this->actor, play, cueChannel);
+    }
+}
+
+void EnAn_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnAn* this = THIS;
+    s32 temp_v1;
+    s32 watchedCouplesMaskCs;
+
+    if (play->sceneId != SCENE_YADOYA) {
+        this->actor.params &= (s16)~ENAN_8000;
+    }
+
+    temp_v1 = (this->actor.params & ENAN_8000) >> 0xF;
```

Comment:
macro?

Reply:
The issue with making a macro for this is this would be the only place where it would be used. The other 4 places where it is used it doesn't shift the param and uses it directly instead

Reply:
I wonder if it would be better to do a `#define ENAN_GET_8000((thisx)->params & ENAN_8000)`, than just include an extra shift here?